python/python/lance/indices.py

```diff
@@ -0,0 +1,253 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright The Lance Authors
+import math
+import warnings
+from typing import TYPE_CHECKING, Optional, Union
+
+import pyarrow as pa
+
+from lance.file import LanceFileReader, LanceFileWriter
+
+from .lance import indices
+
+if TYPE_CHECKING:
+    from .dependencies import torch
+
+
+class IvfModel:
```

Comment:
Not necessary for now, but it would be very cool to be able to get this from an existing index.

Reply:
Agreed.